include/luabind/bind.hpp

```diff
@@ -147,15 +178,21 @@ class class_ {
         //     lua_pushinteger(L, i - 1);
         //     return info->subscript_read(L);
         // }
-        const char* key = lua_tostring(L, 2);
-        auto p_it = info->properties.find(std::string_view(key));
-        if (p_it != info->properties.end()) {
-            return p_it->second.getter(L);
+        if (lua_type(L, 2) == LUA_TNUMBER && info->index_by_number) {
+            return info->index_by_number(L);
         }
-        auto f_it = info->functions.find(std::string_view(key));
-        if (f_it != info->functions.end()) {
-            lua_pushcfunction(L, f_it->second);
-            return 1;
+
+        if (lua_type(L, 2) == LUA_TSTRING) {
+            const char* key = lua_tostring(L, 2);
```

Comment:
Use `value_mirror<std::string_view>::from_lua(L, 2);` it is faster.

Reply:
done